### PR TITLE
Generic theme

### PR DESCRIPTION
**Pull request by adamhstewart**
A theme which is a bit more generic with 
- no color
- no copyrighted fonts
- a bit more appealing header/titles

image

### CODE REVIEW

File: web/themes/custom/server_theme/templates/purgecss.html.twig [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+we are using, so they will not be purged by TailWind's purge CSS when we deoply.
+
+The way CSS purge works is by going over the twig/theme functions and look for
+the class names. If it will see in twig `bg-{{color}}` it will not pick it up,
+as it doesn't know to what it will comiple.
+
+Below are the custom colors we defined in tailwind.config.js
+#}
+
+black
```
Comment (amitaibu) [inline web/themes/custom/server_theme/templates/purgecss.html.twig:11]: This should be moved to tailwind.config.js under the existing `whitelist`

File: web/themes/custom/server_theme/templates/purgecss.html.twig [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+gray-3
+gray-6
+gray-9
+gray-c
+white
+
+grid-cols-cards
```
Comment (amitaibu) [inline web/themes/custom/server_theme/templates/purgecss.html.twig:18]: why is this needed, isn't it auto-picked up from the twig file?

File: web/themes/custom/server_theme/tailwind.config.js [hunk 3]
```diff
@@ -4,8 +4,12 @@ module.exports = {
   theme: {
     extend: {
       colors: {
-        'purple-primary': '#664493',
-        'turquoise': '#03B3AF',
+        'black': '#000000',
+        'gray-3': '#333333',
```
Comment (amitaibu) [inline web/themes/custom/server_theme/tailwind.config.js:8]: since we're removing design, maybe lets use one of the existing grays? https://tailwindcss.com/docs/customizing-colors

File: web/themes/custom/server_theme/tailwind.config.js [hunk 4]
```diff
@@ -4,8 +4,12 @@ module.exports = {
...
+        'gray-6': '#666666',
+        'gray-9': '#999999',
+        'gray-c': '#cccccc',
+        'white' : '#ffffff',
```
Comment (amitaibu) [inline web/themes/custom/server_theme/tailwind.config.js:12]: we already have `text-white` and `text-black` by default - https://tailwindcss.com/docs/text-color

File: web/themes/custom/server_theme/server_theme.libraries.yml [hunk 5]
```diff
@@ -10,6 +10,16 @@ global-scripts:
   dependencies:
     - core/jquery
 
+librefranklin:
```
Comment (amitaibu) [inline web/themes/custom/server_theme/server_theme.libraries.yml:13]: Maybe lets replace with https://fonts.google.com/specimen/Roboto , which seems to be more popular?

File: web/themes/custom/server_theme/src/scss/components/typography.scss [hunk 6]
```diff
@@ -1,17 +1,22 @@
+body {
+  font-family: 'Open Sans', sans-serif;
```
Comment (amitaibu) [inline web/themes/custom/server_theme/src/scss/components/typography.scss:2]: This should be on the twig file

File: web/themes/custom/server_theme/src/scss/components/typography.scss [hunk 7]
```diff
@@ -1,17 +1,22 @@
...
+}
+
 h1, h2, h3, h4 {
-  @apply leading-none text-turquoise;
+  font-family: 'Libre Franklin', sans-serif;
```
Comment (amitaibu) [inline web/themes/custom/server_theme/src/scss/components/typography.scss:6]: This should be changed in tailwind.config.js - we already have https://github.com/Gizra/drupal-starter/blob/bebe2b00a00a228ad0d81b38871def43ae07a7ab/web/themes/custom/server_theme/tailwind.config.js#L29-L30

Rule of thumb is that we always try to stick to `@apply`, thus making sure we have a better consistency, and no CSS rules "in the wild"

**Review by amitaibu — COMMENTED** [review]
Thanks for tackling this! I've added a few comments.

File: web/themes/custom/server_theme/src/scss/components/typography.scss (hunk 6 above)
Comment (adamhstewart) [reply web/themes/custom/server_theme/src/scss/components/typography.scss:2]: or `@apply font-body`?

File: web/themes/custom/server_theme/templates/purgecss.html.twig (hunk 1 above)
Comment (adamhstewart) [reply web/themes/custom/server_theme/templates/purgecss.html.twig]: @amitaibu is the purgecss file an outdated method? - if so I'll remove it from other projects too.

**Comment by adamhstewart** [comment]
@amitaibu I think I hit everything - ready for round 2 at your convenience.

File: web/themes/custom/server_theme/src/scss/components/typography.scss (hunk 6 above)
Comment (amitaibu) [reply web/themes/custom/server_theme/src/scss/components/typography.scss]: I prefer to add a CSS rule only when it's too complicated to add it to twig. What's nice about TW is that by reading a twig file, I know how it will behave

File: web/themes/custom/server_theme/templates/purgecss.html.twig (hunk 1 above)
Comment (amitaibu) [reply web/themes/custom/server_theme/templates/purgecss.html.twig]: > is the purgecss file an outdated method?

Yes. `whitelist` is the right way

File: web/themes/custom/server_theme/src/scss/components/typography.scss (hunk 6 above)
Comment (adamhstewart) [reply web/themes/custom/server_theme/src/scss/components/typography.scss]: Ah, I see on the body tag in particular....Headers, we still want to declare in CSS so it applies over all uses (like sizes)?

Comment (adamhstewart) [reply web/themes/custom/server_theme/src/scss/components/typography.scss]: or better even? https://tailwindcss.com/docs/adding-base-styles#using-a-plugin

File: web/themes/custom/server_theme/tailwind.config.js [hunk 8]
```diff
@@ -1,12 +1,14 @@
 const plugin = require('tailwindcss/plugin');
+const colors = require('tailwindcss/colors')
 
 module.exports = {
   theme: {
+    colors: {
```
Comment (amitaibu) [inline web/themes/custom/server_theme/tailwind.config.js:6]: I think we can remove this part entirely, and have all TW's default colors loaded by default

File: web/themes/custom/server_theme/src/scss/components/typography.scss (hunk 6 above)
Comment (amitaibu) [reply web/themes/custom/server_theme/src/scss/components/typography.scss]: > or better even? https://tailwindcss.com/docs/adding-base-styles#using-a-plugin

In the case of h1-h5 I'd keep it in the scss file.

With that said, after reading the Refactoring UI book, I understand that giving h1 the biggest size, and h5 the lowest is most probably not the right way.

Comment (adamhstewart) [reply web/themes/custom/server_theme/src/scss/components/typography.scss]: > I understand that giving h1 the biggest size, and h5 the lowest is most probably not the right way.

never was... Every homepage ever is an example of this. IMO sizes should live in the templates.

File: web/themes/custom/server_theme/server_theme.theme [hunk 9]
```diff
@@ -141,3 +148,12 @@ function server_theme_preprocess_page_title(&$variables) {
     $variables['title'] = '';
   }
 }
+
+/**
+ * Implements hook_theme_suggestions_HHOK_alter().
```
Comment (amitaibu) [inline web/themes/custom/server_theme/server_theme.theme:153]:
```suggestion
 * Implements hook_theme_suggestions_page_title_alter().
```

File: web/themes/custom/server_theme/templates/server-theme-page-title.html.twig [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+ *   displayed in front of the main title tag that appears in the template.
+ * - title: The page title, for use in the actual content.
+ * - title_suffix: Additional output populated by modules, intended to be
+ *   displayed after the main title tag that appears in the template.
+ */
+#}
+{%
+  set classes = [
+    'page-title',
+    'text-5xl',
```
Comment (amitaibu) [inline web/themes/custom/server_theme/templates/server-theme-page-title.html.twig:18]: Maybe 

```suggestion
    'text-lg',
```

**Comment by amitaibu** [comment]
Lovely, thanks.
